File: jpa-fetch-join/src/main/java/com/example/jpafetchjoin/address/Address.java
package com.example.jpafetchjoin.address;


import com.example.jpafetchjoin.order.Order;
import com.fasterxml.jackson.annotation.JsonIgnore;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "address")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class Address {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private long id;

  @Column(name = "citiy", nullable = false)
  private String citiy;

  @ManyToOne
  @JoinColumn(name = "order_id", nullable = false)
  @JsonIgnore
  private Order order;

  public Address(String citiy, Order order) {
    this.citiy = citiy;
    this.order = order;
    this.order.addAddress(this);
  }

}

File: redis/src/main/kotlin/com/example/redis/Address.kt
package com.example.redis

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.Table
import org.springframework.data.jpa.repository.JpaRepository

@Entity(name = "address")
@Table(name = "address")
class Address(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    val id: Long? = null,

    @Column(name = "street", length = 100, nullable = false)
    val street: String,

    @Column(name = "city", length = 50, nullable = false)
    val city: String,

    @Column(name = "state", length = 50, nullable = false)
    val state: String,

    @Column(name = "zipcode", length = 10, nullable = false)
    val zipcode: String
)

interface AddressRepository : JpaRepository<Address, Long>